module-system/module-implementations/sov-attester-incentives/src/query.rs

```diff
@@ -44,22 +44,11 @@ where
         }
     }
 
-    /// Used by attesters to get a proof that they were bonded before starting to produce attestations.
-    /// A bonding proof is valid for `max_finality_period` blocks, the attester can only produce transition
-    /// attestations for this specific amount of time.
-    pub fn get_bond_proof(
```

Comment:
@preston-evans98 how important is it for actual users?

Could we provide this functionality just from witness?

Reply:
Another solution would be to generalize it full node somehow, so you can take proof for any prefix, then clients just have to query storage prefix for address (or build it themself).

Or it should be part of WorkingSet API

Reply:
I need to investigate if I making this part of WorkingSet will solve the problem.

Reply:
Did it differently